presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java

```diff
@@ -566,6 +566,54 @@ else if (activeCentroids == 1) {
         return weightedAverage(mean[n - 1], z1, max, z2);
     }
 
+    public double trimmedMean(double l, double h)
+    {
+        checkArgument(l >= 0 && l <= 1, "l should be in [0,1], got %s", l);
+        checkArgument(h >= 0 && h <= 1, "h should be in [0,1], got %s", h);
+
+        if (unmergedWeight > 0) {
+            compress();
+        }
+
+        if (activeCentroids == 0 || l >= h) {
```

Comment:
Should l >= h be an error instead?  Returning NaN seems somewhat counter-intuitive for that, I would assume it would either swap them or error.

Reply:
I'm not sure what makes the most sense here. While l and h outside [0, 1] really doesn't make sense, l > h could potentially be interpreted as "taking the average over an empty portion of the distribution", then it would make sense to return NaN. But I can change it if you think it's more intuitive to throw an error.
Also, I'm not sure what we should return if l == h. We could either return the mean[i] where the bounds are (the first if case), or have it error out or return NaN.

Reply:
For l == h I think it could also just return cdf(l) (or NaN).  I think this is probably fine to submit like this for now and see what the maintainers say.